types/helpers/helpers.core.d.ts

```diff
@@ -0,0 +1,136 @@
+/**
```

Comment:
Some of the type checkers in here might benefit from include the `value is <x>` part like `isNumber` in the example here 
 https://www.typescriptlang.org/docs/handbook/advanced-types.html#typeof-type-guards in the definitions